packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -749,4 +767,37 @@ export const MenuBarMixin = (superClass) =>
         this._subMenu.close();
       }
     }
+
+    /** @private */
+    _requestClose() {
+      this._preventClose = false;
+      setTimeout(() => {
+        if (!this._preventClose) {
+          this._close(false);
+        }
+      }, this.hideDelay);
+    }
+
+    /** @private */
+    _addHoverListener(subMenu) {
+      const menuOverlay = subMenu.$.overlay.$.overlay;
+
+      menuOverlay.addEventListener('mouseleave', () => {
+        if (this.openOnHover) {
```

Comment:
All the tests pass with this check removed. Needs a test where an opened menu bar doesn't auto-close if its `openOnHover` is set to `false`.